### ISSUE TITLE
Error thrown

### ISSUE THREAD

**Issue by csga5000** [EntropyString/JavaScript#2]
I've got an angular application.

I added your library to my package.json:
```
"entropy-string": "^4.0.3"
```

I ran npm, install and in my ts file:
```
import { Entropy, charset32 } from 'entropy-string';
...
var str = new Entropy({ total: 1e5, risk: 1e9, charset: charset32}).string();
```

This compiles file, but throws an exception at runtime when the code is executed:
```
ERROR TypeError: Crypto.randomBytes is not a function
    at csprngBytes (entropy.js:89)
    at _class.string (entropy.js:245)
    at EditUserDialog.genPassword (admin.component.ts:167)
    at Object.eval [as handleEvent] (EditUserDialog.html:10)
    at handleEvent (core.js:13581)
    at callWithDebugContext (core.js:15090)
    at Object.debugHandleEvent [as handleEvent] (core.js:14677)
    at dispatchEvent (core.js:9990)
    at eval (core.js:10611)
    at HTMLElement.eval (platform-browser.js:2628)
```

Not sure why.  window.crypto only has a "getRandomValues" function, and your dependencies don't list any sort of crypto package.  I tried instally crypto, and crypto-js just to see if that'd fix it, but it didn't.

**Comment by dingosky** [comment]
For environments that don't have **crypto**, you can use the PRNG `Math.random` by passing `prng: true` in the constructor. For your example above:

`var str = new Entropy({ total: 1e5, risk: 1e9, charset: charset32, prng: true}).string();`

**Comment by csga5000** [comment]
The entire point of this package is to allow securely generating passwords stuff, doing that would defeat the purpose.  And the browser allows generating cryptographically random bytes, so I don't see why this couldn't work.

Here for example is the temporary code I wrote (I don't understand byte to string conversion well enough to know if this is a good approach, but I intended to use it only temporarily in a software that is currently in a very small beta test stage:

```
	genPassword() {
		var array = new Uint32Array(2);
  		window.crypto.getRandomValues(array);
		this.user.password = '';
		for (var i = 0; i < array.length; i++)
			this.user.password += array[i].toString(32);
		this.genPass = this.user.password;
	}
```

**Comment by dingosky** [comment]
You can pass the bytes to use via the `stringWithBytes` method. The number of bytes needed to produce a string is given by the `bytesNeeded` method.

```js
import { Entropy, charset32 } from 'entropy-string';

const entropy = new Entropy({ total: 1e5, risk: 1e9, charset: charset32);
const bytesNeeded = entropy.bytesNeeded();

let bytes = new Uint8Array(bytesNeeded);
window.crypto.getRandomValues(bytes);
const string = entropy.stringWithBytes(bytes);
```

The suggestion of using a PRNG was without knowing your use case. Cryptographically strong bytes are important enough they are the default, but that's not the only use of the package. Random ids often don't really need a CSPRNG even though we tend to go that route anyway. In general, using a CSPRNG does add a failure possibility (exhausted entropy source) that does not plague PRNG usage. According to the [docs](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues), `window.crypto.getRandomValues` avoids that issue by seeding a PRNG with "enough" entropy to be _suitable for cryptographic usages_. I'm not familiar with the metric they use to determine what enough seeding entropy actually is.

**Comment by csga5000** [comment]
Okay, thank you that is helpful, as I assume it probably won't crash that way.

I originally assumed this was just targetting "javascript" not nodejs, but as I get an exception running example code in both firefox and chrome is it the case that this library is actually targetting nodejs?  If so, I think it'd at least be helpful to those who may come across this package if somewhere in your readme you mentioned that this library is for nodejs, or that you will have to provide bytes if you intend to use it in javascript. (Or just make it use the crypto.getRandomValues() for javascript so it supports both without exceptions).

**Comment by dingosky** [comment]
In version __4.1.0__, I've restructured the package and created a browser version in the file __entropy-string.browser.js__. As noted in the updated README, the browser version uses `window.crypto.getRandomValues`. There is simple example usage in the file __examples/browser.html__.

**Comment by csga5000** [comment]
Awesome, well I'll close this for you then, since that addresses this pretty well I would think

**Comment by dingosky** [comment]
Fixed in v4.1.0
